packages/clay-color-picker/src/Custom.js

```diff
@@ -0,0 +1,245 @@
+				fill="#6B6C7E"
+			/>
+		</svg>
+	);
+}
+
+RGBInput.propTypes = {
+	onChange: PropTypes.func,
+	name: PropTypes.string,
+	value: PropTypes.number,
+};
+
+/**
+ * Renders input that displays RGB values
+ * @return {React.Component}
+ */
+function RGBInput({onChange, name, value}) {
+	return (
+		<div className="form-group rgb-info">
+			<div className="input-group">
+				<div className="input-group-item input-group-item-shrink input-group-prepend">
+					<span className="input-group-text">
+						{name.toUpperCase()}
+					</span>
+				</div>
+				<div className="input-group-append input-group-item">
+					<input
+						value={value}
+						className="form-control"
+						type="text"
+						onChange={event => {
+							const newVal = Number(event.target.value);
+
+							onChange({[name]: newVal});
+						}}
+					/>
+				</div>
+			</div>
+		</div>
+	);
+}
+
+Custom.propTypes = {
+	colors: PropTypes.arrayOf(PropTypes.string),
+	label: PropTypes.string,
+	onChange: PropTypes.func,
+	onColorsChange: PropTypes.func,
+	value: PropTypes.string,
+};
+
+/**
+ * Renders the custom color picker
+ * @return {React.Component}
+ */
+function Custom({colors, label, onChange, onColorsChange, value}) {
+	const color = tinycolor(value);
+
+	const [activeSplotchIndex, setActiveSplotchIndex] = useState(0);
+	const [hue, setHue] = useState(0);
+	const [editorActive, setEditorActive] = useState(false);
+	const [inputVal, setInputValue] = useState(color.toHex());
+
+	const {r, g, b} = color.toRgb();
+	const {s, v} = color.toHsv();
+
+	const rgbArr = [[r, 'r'], [g, 'g'], [b, 'b']];
+
+	const handleNewInputValue = value => {
+		const match = value.match(HEX_REGEX);
+
+		setInputValue(match ? match[0] : '');
+	};
+
+	const setNewColor = (colorValue, setInput = true) => {
+		const hexString = colorValue.toHexString();
+
+		const newColors = [...colors];
+
+		newColors[activeSplotchIndex] = hexString;
+
+		onColorsChange(newColors);
+
+		onChange(hexString);
+
+		if (setInput) {
+			handleNewInputValue(colorValue.toHex());
+		}
+	};
+
+	return (
+		<div>
+			{label && (
+				<div className="label-container">
+					<label>{label}</label>
+					<button
+						onClick={() => setEditorActive(!editorActive)}
+						className={`${
+							editorActive ? 'active ' : ''
+						}btn btn-monospaced btn-sm`}
+						type="button"
+					>
+						<CustomColorIcon />
+					</button>
+				</div>
+			)}
+
+			<div className="splotch-grid">
+				{colors.map((hex, i) => (
+					<Splotch
+						active={i === activeSplotchIndex}
+						onClick={() => {
+							if (hex === '#FFFFFF') {
+								setEditorActive(true);
+							}
+
+							setActiveSplotchIndex(i);
+
+							setHue(tinycolor(hex).toHsv().h);
+
+							onChange(hex);
+						}}
+						key={i}
+						value={hex}
+					/>
+				))}
+			</div>
+
+			{editorActive && (
+				<React.Fragment>
```

Comment:
Ideally we'd use the shorthand here:

```javascript
<>
  ...
</>
```

In practice, you only need to use `React.Fragment` if some piece of your build pipeline (lints, Babel etc) is old and doesn't handle the shorthand syntax. Not sure if there's anything in the package.json here that would prevent this here.

Reply:
Since we are still on babel 6, we have to use the more explicit form of Fragment.